File: backend/src/main/java/by/fpmibsu/findafriend/controller/commands/adverts/CreateAdvertHandler.java
package by.fpmibsu.findafriend.controller.commands.adverts;

import by.fpmibsu.findafriend.application.authentication.AuthenticationData;
import by.fpmibsu.findafriend.application.mediatr.RequestHandler;
import by.fpmibsu.findafriend.dataaccesslayer.advert.AdvertDao;
import by.fpmibsu.findafriend.entity.Advert;
import by.fpmibsu.findafriend.entity.Place;
import by.fpmibsu.findafriend.entity.User;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Date;
import java.time.Instant;

public class CreateAdvertHandler extends RequestHandler<Boolean, CreateAdvertCommand> {
    private final Logger logger = LogManager.getLogger(CreateAdvertHandler.class);
    private final AdvertDao advertDao;
    private final AuthenticationData authenticationData;

    public CreateAdvertHandler(AdvertDao advertDao, AuthenticationData authenticationData) {
        this.advertDao = advertDao;
        this.authenticationData = authenticationData;
    }

    @Override
    public Boolean handle(CreateAdvertCommand request) {
        var place = new Place();
        place.setId(request.placeId);
        var u = new User();
        u.setId(Integer.parseInt(authenticationData.getClaim("id")));
        var advert = new Advert(0, request.title, request.description, Date.from(Instant.now()), place, u, Advert.AdvertType.fromValue(request.advertType));
        advertDao.create(advert);
        return true;
    }
}

File: BackendNET/Backend/Commands/Adverts/CreateAdvert/CreateAdvertHandler.cs
using System.Data.Common;
using Backend.Entities;
using Backend.Repository;
using MediatR;

namespace Backend.Commands.Adverts.CreateAdvert;

public class CreateAdvertHandler : IRequestHandler<CreateAdvertCommand, bool>
{
    private readonly IAdvertRepository _repository;

    public CreateAdvertHandler(IAdvertRepository repository)
    {
        _repository = repository;
    }

    public async Task<bool> Handle(CreateAdvertCommand request, CancellationToken cancellationToken)
    {
        var instance = new Advert {
            CreationDate = DateTime.Now,
            Description = request.Description,
            OwnerId = request.UserId,
            PlaceId = request.PlaceId,
            Title = request.Title,
            Type = request.AdvertType
        };

        try
        {
            await _repository.InsertAsync(instance, cancellationToken);
            return true;
        }
        catch (DbException e)
        {
            return false;
        }
    }
}